redbot/core/events.py

```diff
@@ -241,10 +243,116 @@ async def on_command_error(ctx, error, unhandled_by_cog=False):
+                    )
+                elif error.minimum is None and error.maximum is not None:
+                    message = _(
+                        "Provided value `{value}` for argument `{parameter_name}` must be an integer no more than {maximum}."
+                    )
+                elif error.minimum is not None and error.maximum is None:
+                    message = _(
+                        "Provided value `{value}` for argument `{parameter_name}` must be an integer no less than {maximum}."
+                    )
+                elif error.maximum is not None and error.minimum is not None:
+                    message = _(
+                        "Provided value `{value}` for argument `{parameter_name}` must be an integer between {minimum} and {maximum}."
+                    )
+            elif isinstance(error.value, float):
+                value = error.value
+                if (error.minimum == 0 or error.minimum == 1) and error.maximum is None:
+                    message = _(
+                        "Provided value `{value}` for argument `{parameter_name}` must be a positive number."
+                    )
+                elif error.minimum is None and error.maximum is not None:
+                    message = _(
+                        "Provided value `{value}` for argument `{parameter_name}` must be a number no more than {maximum}."
+                    )
+                elif error.minimum is not None and error.maximum is None:
+                    message = _(
+                        "Provided value `{value}` for argument `{parameter_name}` must be a number no less than {maximum}."
+                    )
+                elif error.maximum is not None and error.minimum is not None:
+                    message = _(
+                        "Provided value `{value}` for argument `{parameter_name}` must be a number between {minimum} and {maximum}."
+                    )
+            elif isinstance(error.value, str):
+                count = len(error.value)
+                if count == 1:
+                    value = _("one character")
+                else:
+                    value = _("{count} characters").format(count=count)
+                if error.minimum is None and error.maximum is not None:
+                    message = _(
+                        "Provided value `{value}` for argument `{parameter_name}` must be a string no more than {maximum}."
+                    )
+                elif error.minimum is not None and error.maximum is None:
+                    message = _(
+                        "Provided value `{value}` for argument `{parameter_name}` must be a string no less than {minimum}."
+                    )
+                elif error.maximum is not None and error.minimum is not None:
+                    message = _(
+                        "Provided value `{value}` for argument `{parameter_name}` must be a string between {minimum} and {maximum}."
```

Comment:
```suggestion
                        "Provided value `{value}` for argument `{parameter_name}` must be a string with a length of no more than {maximum}."
                    )
                elif error.minimum is not None and error.maximum is None:
                    message = _(
                        "Provided value `{value}` for argument `{parameter_name}` must be a string with a length of no less than {minimum}."
                    )
                elif error.maximum is not None and error.minimum is not None:
                    message = _(
                        "Provided value `{value}` for argument `{parameter_name}` must be a string with a length of between {minimum} and {maximum}."
```

These strings should be changed for when the Range annotation is `str`, because a "string no more than [...]" isn't grammatically fitting. I've made a suggestion but feel free to improvise if you have a better idea.